### ISSUE TITLE
Improve output on 404 and recreate the resource

### ISSUE THREAD

**Issue by alexandrujieanu** [Ouest-France/terraform-provider-phpipam#18]
Hello,

I'm using `master` branch and terraform 0.12.24.
When the IPAM entry gets deleted terraform returns:

```
Error: Error from API (404): Invalid Id
```

I think in case of 404 the provider terraform should proceed to recreate the entry.
Maybe you can do something about this.

More information below:

Using `TF_LOG=ERROR` points to the ipam terraform module, but again not specifically to a particular IPAM entry. Each vm has multiple interfaces and IPAM entries.

```
2020/06/03 15:21:09 [ERROR] module.vm.module.ipam: eval: *terraform.EvalRefresh, err: Error from API (404): Invalid Id
2020/06/03 15:21:09 [ERROR] module.vm.module.ipam: eval: *terraform.EvalSequence, err: Error from API (404): Invalid Id
```

Currently the only option I'm aware of is to identify which resource failed (which entry was deleted from IPAM) and do a `state rm` on it.

Thanks!

**Comment by pablo-ruth** [comment]
Hello,

You are right, as specified by Terraform docs (https://www.terraform.io/docs/extend/writing-custom-providers.html#implementing-read) the provider must remove the resource from the state when not found, I fixed this in PR #19 and released a new version v0.4.2 .

**Comment by alexandrujieanu** [comment]
Thank you!
